src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/CodeCache.java

```diff
@@ -104,27 +86,22 @@ public CodeBlob findBlobUnsafe(Address start) {
       return null;
     }
 
-    try {
-      result = (CodeBlob) virtualConstructor.instantiateWrapperFor(containing_heap.findStart(start));
-    }
-    catch (WrongTypeException wte) {
-      Address cbAddr = null;
-      try {
-        cbAddr = containing_heap.findStart(start);
-      }
-      catch (Exception findEx) {
-        findEx.printStackTrace();
-      }
+    Address cbAddr = containing_heap.findStart(start);
+    if (cbAddr == null) return null;
 
+    return createCodeBlobWrapper(cbAddr, start);
+  }
+
+  public CodeBlob createCodeBlobWrapper(Address cbAddr, Address start) {
```

Comment:
I think the use of the name "start" here is a carryover from `findBlobUnsafe(Address start)`. I find it a very misleading name. cbAddr points to the "start" of the blob. "start" points somewhere in the middle of the blob. In fact callers of this API somethimes pass in findStart(addr) for cbAddr, which just adds to the confusion. Perhaps this is a good time to rename "start" to something else, although I can't come up with a good suggestion, but I think anything other than "start" would be an improvement. Maybe "pcAddr".  That aligns with the "for PC=" message below. Or maybe just "ptr" which aligns with `createCodeBlobWrapper(findStart(ptr), ptr);`

Reply:
`cbPc` with comment explaining that it could be inside code blob.

Reply:
That sounds fine.

Reply:
done